### PR TITLE
holster code refactor and additions

### PR DESCRIPTION
**Pull request by ErisMeleeEnjoyer**
## About The Pull Request

adds holsters to loadout menu: pocket, baton, sheath, belt, concealed carry, scabbard

moves all holsters in holster.dm file, since all of them are put into one type of clothing for simplicity and so that the holster hotkey could work

Baton holster can now hold hatchets, hammers and its subtypes, as long as they are ITEM_SIZE_BULKY or less.(#7574)

changes holster verb to take implicit source instead of explicit (src = usr.contents instead of src in usr)

gives the holster verb a set of priorities to take guns from if you have multiple holsters; if using the verb with an empty hand, you draw the gun from your holsters, otherwise you try to put the item in your holsters.

removes bay12 holster accessory code(it was godawful), refactors it completely, turns bay12 holsters into concealed carry holsters that **fit the same things as before(normal-sized handguns)**, and maybe even pizza slices

resprites bay12 accessory holster

image

## Why It's Good For The Game

you won't have to give a fluoride stare to your opponents in the middle of a fight when you have two belt holsters

you actually can have multiple holsters now

you actually can use pouch holsters now

bay12 holsters are not nerfed in this one(if someone doesn't read the PR's description and comments about nerfing, I will have no choice but to post the image)

## Changelog
:cl: Kegdo
add: belt holster
add: accessory holster resprite
del: bay12 godawful holster sprites and code
tweak: holster pouches moved into one subtype for hotkey use
code: holster code refactor
/:cl:

### CODE REVIEW

**Comment by ErisMeleeEnjoyer** [comment]
in case someone still doesn't believe muh "good intentions"
image

**Comment by ErisMeleeEnjoyer** [comment]
image

File: code/game/objects/items/weapons/storage/holster.dm [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+		/obj/item/gun/energy/retro/sawn,
+		/obj/item/gun/projectile/automatic/luty,
+		/obj/item/gun/projectile/revolver/hornet
+		)
+
+	sliding_behavior = TRUE
+
+/obj/item/storage/pouch/holster/baton
+	name = "baton sheath"
+	desc = "Can hold a baton, or indeed most weapon shafts."
```
Comment (hyperioo) [inline code/game/objects/items/weapons/storage/holster.dm:51]:
```suggestion
	desc = "Can hold a baton and most weapon shafts."
```

**Review by hyperioo — APPROVED** [review]

**Comment by ValoTheValo** [comment]
> in case someone still doesn't believe muh "good intentions" image

well you still ended up nerfing them by removing quickdraw

**Comment by assortedbeads** [comment]
> > in case someone still doesn't believe muh "good intentions" image
> 
> well you still ended up nerfing them by removing quickdraw. cant draw guns out of my holster now. 
> 
> also the new sprite looks kinda bad with most suits

This, it's only working on the pocket ones

**Comment by assortedbeads** [comment]
> > > in case someone still doesn't believe muh "good intentions" image
> > 
> > well you still ended up nerfing them by removing quickdraw. cant draw guns out of my holster now. 
> > 
> > also the new sprite looks kinda bad with most suits
> 
> This, it's only working on the pocket ones

Didn't think we would have to circle back to the issue but here we are.

Main complaint on the pr before the test merge was that it removed the bay holsters, and people wanted them back with their original funtionality.

Coder pretended to understand the issue and claimed they had fixed it, only to strip the quickdraw functionality and give it to their new holsters, making the uniform holsters work like pockets.

Apparently it is "no issue" and people that say so are just "seething and coping."

**Comment by maikilangiolo** [comment]
Good change. There's now a reason to use the non-accessory holsters, and you can still holster what you want in your suit.

**Pull request merged**